### PR TITLE
Fix s3Cluster schema inference in parallel distributed insert select

### PR DESCRIPTION
**Pull request by Avogar**
### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in official stable or prestable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix s3Cluster schema inference in parallel distributed insert select when `parallel_distributed_insert_select` is enabled

### CODE REVIEW

**Review by nikitamikhaylov — APPROVED** [review]

**Review by alexey-milovidov — APPROVED** [review]

**Comment by alexey-milovidov** [comment]
The test appeared flaky.

**Comment by Avogar** [comment]
> The test appeared flaky.

Investigating

File: tests/integration/test_s3_cluster/test.py [hunk 1]
```diff
@@ -326,3 +326,45 @@ def test_distributed_insert_select_with_replicated(started_cluster):
...
+    ) SETTINGS parallel_distributed_insert_select=1, use_structure_from_insertion_table_in_table_functions=0;
+        """
+    )
+
+    actual_count = int(
+        node.query(
+            "SELECT count() FROM s3('http://minio1:9001/root/data/generated/*.csv', 'minio', 'minio123', 'CSV','a String, b UInt64')"
+        )
+    )
+
```
Comment (nikitamikhaylov) [inline tests/integration/test_s3_cluster/test.py:360]: Probably adding SYSTEM SYNC REPLICA would be better here?

**Comment by antonio2368** [comment]
`test_parallel_distributed_insert_select_with_schema_inference` became flaky after this

https://play.clickhouse.com/play?user=play#c2VsZWN0IApjaGVja19zdGFydF90aW1lIGFzIGQsIHB1bGxfcmVxdWVzdF9udW1iZXIsIHJlcG9ydF91cmwKZnJvbSBjaGVja3MKd2hlcmUKICAgICcyMDIzLTAxLTAxJyA8PSBjaGVja19zdGFydF90aW1lIGFuZCB0ZXN0X25hbWUgbGlrZSAnJXRlc3RfcGFyYWxsZWxfZGlzdHJpYnV0ZWRfaW5zZXJ0X3NlbGVjdF93aXRoX3NjaGVtYV9pbmZlcmVuY2UlJyAKICAgIGFuZCB0ZXN0X3N0YXR1cyBpbiAoJ0ZBSUwnLCAnRkxBS1knLCAnRVJST1InKSAKb3JkZXIgYnkgZCBkZXNj

**Comment by Avogar** [comment]
I managed to reproduce flakiness locally. Let me investigate it

**Comment by Avogar** [comment]
Let's try https://github.com/ClickHouse/ClickHouse/pull/46488.
Locally I cannot reproduce it abymore after this fix. I have a hypothesis why it's flaky, but I am not sure

**Pull request merged**